packages/flutter_tools/lib/src/project.dart

```diff
@@ -39,6 +39,27 @@ enum SupportedPlatform {
   root, // Special platform to represent the root project directory
 }
 
+String? platformToSubdirectoryPrefix(SupportedPlatform platform) {
```

Comment:
Surprised we don't already have this somewhere.  You could instead change `SupportedPlatform` to an enhanced enum:
```dart
enum SupportedPlatform {
  android('android'),
  ios('ios'),
  linux('linux'),
  macos('macos'),
  web('web'),
  windows('windows'),
  fuchsia('fuchsia'),
  root('root'); // Special platform to represent the root project directory

  const SupportedPlatform(this.subdirectory);
  final String subdirectory;
}
```
Also, what makes it a prefix?